### PR TITLE
Buff power armor

### PR DESCRIPTION
**Pull request by NBBloodhound**
#### Summary

SUMMARY: Content "Modernize & buff power armor"

#### Purpose of change

while its immense protection is the star of the show of course, it's very challenging to use, especially considering the hauling frame can't hold a damned rifle, as well as the fact that the weight makes it unusable for a lot of survivors, even though the description suggests this wouldn't be the case, and, to top it all off, it lacks utility, while it shouldn't be as good as the RM13 or PIS in that category, it still feels a bit lacking, as the description suggests it has some form of climate control. while i don't know how to solve the weight issue, i did make using power armor a hell of a lot easier
Describe the solution

I gave power armor pockets for 4 magazines, 3 nades, and a large rifle, and the heavy suit has a mount flag, because big armor and big gun go together like cheese & wine, gave the hauling frame a more lenient length limit, gave the light helmet the swim_goggles property, and gave all activated power armor climate control
Describe alternatives you've considered

give the hauling frame the nade and mag pockets instead of the suit itself

#### Testing

 game doesn't crash on load!
 power armor fits barrett m107a1, 4 mags for it, and 9 grenades (oops) EDIT: Fixt, down to 3
 turning on doesn't delete items
 reloading gun can't pull from power armor, might have something to do w/ pocket type? more experimentation required, is known bug

#### Additional context

i'll eventually figure out how to make stuff give a strength bonus, as well as the light some dive bonuses, to help it fill the tactical niche better, lmk if you know how to do that

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
I really doubt a person trapped in a power armor has any possibility to use pockets.

**Comment by RobertoVGoulart** [comment]
> I really doubt a person trapped in a power armor has any possibility to use pockets.

If you have enough hand dexterity to reload a gun, you are able to take an item out of a pocket, or some kind of elastic band holding said item, on the armor.

**Comment by mlangsdorf** [comment]
The idea is that the "pockets" are really clamps or other PA accessible storage spaces.  The old style PA that couldn't carry *anything* doesn't make sense - the military would want a MOLLE equivalent of easily accessible ammunition and consumables, and that's why I suggested this.

**Comment by mlangsdorf** [comment]
I fixed the spaces in the summary line so the validator would pass it.
The build errors and test case failures seem to be random failures unrelated to this PR.

File: data/json/items/armor/power_armor.json [hunk 1]
```diff
@@ -131,6 +267,7 @@
         "pocket_type": "CONTAINER",
         "rigid": true,
         "max_contains_volume": "30 L",
+        "max_contains_length": "40 cm",
```
Comment (kevingranade) [inline data/json/items/armor/power_armor.json:270]:
```suggestion
        "max_item_length": "40 cm",
```

**Comment by Zireael07** [comment]
@mlangsdorf: Can that bit about clamps be called out in item description or somewhere in code, so that further contributors don't get confused by power armor apparently having pockets?

**Comment by mlangsdorf** [comment]
I think updating the description to add the fact that it has clamps make sense.  I'm not the author, though.

**Comment by NBBloodhound** [comment]
@mlangsdorf @Zireael07  i agree, i'll do some fiddling w/ the description, when i get around to it, any ideas? im thinking something along the lines of `The armor is outfitted with a variety of clips & clamps, allowing it to hold a few mags & grenades, as well as a large rifle`

**Comment by I-am-Erk** [comment]
I don't think the climate control addition makes a lot of sense here. These aren't fallout power armour, they're hauling frames with armour plating added.

**Comment by NBBloodhound** [comment]
@I-am-Erk true, but from the bodyglove thing, i assumed there was a simple climate control system to prevent, if nothing else, you from baking inside the armor in hot conditions, and keeping the glass warm when it's cold. however, if you so desire, i could get rid of the climate control, but i might consider adding a whole new power armor optimized for use with flamethrowers and other fire weapons 0u0

**Comment by I-am-Erk** [comment]
I don't think power armour really reached the stage for that kind of specialization.

The problem is mostly that climate control is too strong. It wouldn't be unreasonable for it to have a bit of cooling in hot weather, but it's not likely to be keeping you all that comfy in a migo base for example.

**Comment by NBBloodhound** [comment]
@I-am-Erk during my testing, i brought it into a mi-go tower, even with the assistance of ICC, it wasn't sufficient, IIRC
with how warm power armor is, the climate control was sufficient to keep temp neutral, with a bit of flex. i bet it'd be pushed to its limit at ~90 degrees F ambient

**Comment by I-am-Erk** [comment]
Well, if it's that weak already it might work then

**Comment by anothersimulacrum** [comment]
Looks to just be waiting on a merge.

**Pull request merged**